lib/capybara/selector.rb

```diff
@@ -380,15 +381,21 @@
   end
 
   filter(:selected) do |node, selected|
-    actual = node.all(:xpath, './/option', visible: false).select { |option| option.selected? }.map { |option| option.text(:all) }
-    [selected].flatten.sort == actual.sort
+    actual = node.all(:xpath, './/option', visible: false).select(&:selected?).map { |option| option.text(:all) }
+    Array(selected).sort == actual.sort
```

Comment:
Array(selected) and [selected].flatten are not identical since the latter will flatten multiple nested arrays - In this instance it probably doesn't make any difference though